AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -357,6 +366,13 @@ - (void)performBatchUpdates:(void (^)())updates completion:(void (^)(BOOL))compl
   [self performBatchAnimated:YES updates:updates completion:completion];
 }
 
+- (void)registerSupplementaryNodeOfKind:(NSString *)elementKind
+{
+  [_registeredSupplementaryKinds addObject:elementKind];
```

Comment:
Add an assert and runtime safety check for nil (and zero-length?  maybe that's valid?)